app/grandchallenge/components/tasks.py

```diff
@@ -603,19 +606,30 @@ def get_model_instance(*, app_label, model_name, **kwargs):
     return model.objects.get(**kwargs)
 
 
-@acks_late_2xlarge_task
+@acks_late_2xlarge_task(retry_on=(LockNotAcquiredException,))
+@transaction.atomic
```

Comment:
I think that the `raise` in L651 will cause the transaction to be cancelled and the job left in an undefined state.

Reply:
The pitch mentions that all celery tasks involved in job creation should be reviewed and adjusted to fit our new design principles. I can undo the last commit I made and get rid of all those changes. Can indeed be done in a separate PR.

Reply:
And you're right, that the transaction would prevent the intermediate status change from ever taking effect. I think that is ok, to be honest, but maybe better to leave as is for now and make the celery task changes separately.

Reply:
Indeed, job creation, not job execution. Should have been clearer. I know that part of the codebase is tricky and problematic but we really shouldn't touch it given the demand in production right now.